### PR TITLE
feat(file-service): send additional form data with TdFileService 

### PR DESCRIPTION
**Pull request by fpeterdev**
## Description

This is a feature requested from #546. (I created the issue but another account)

With a new optional  `formData` property in `IUploadOptions` we can send additional form data when using `TdFileService`.
There were no tests for the `TdFileService` so I created two test cases: one with additional headers and data and one without.

### What's included?

- new `formDat` property to `IUploadOptions`
- add elements to `formData` in the service
- new tests for `TdFileService`
- update for README.md file

#### General Tests for Every PR

- [ ] `ng serve --aot` still works.
I got a JavaScript heap out of memory error just as described in this [issue](https://github.com/angular/angular-cli/issues/5618)
- [x] `npm run lint` passes.
- [x] `npm test` passes and code coverage is not lower.
- [x] `npm run build` still works.

### CODE REVIEW

**Comment by emoralesb05** [comment]
Awesome! Thanks for the PR! Will have the team check it out.

File: src/platform/core/file/file-upload/README.md [hunk 1]
```diff
@@ -83,14 +83,15 @@ export class MyModule {}
...
+Recieves as parameter an object that implements the [IUploadOptions] interface. *Note: since the file is sent by the name 'file' you can't use the string 'file' as a key in [additionalFormData]*
 
 ```typescript
 interface IUploadOptions { 
   url: string; 
   method: 'post' | 'put'; 
   file: File;
-  headers?: {[key: string]: string} 
+  headers?: {[key: string]: string};
+  additionalFormData?: {[key: string]: string};
```
Comment (emoralesb05) [inline src/platform/core/file/file-upload/README.md]: Would it be better to make it work as one or the other? 

Like If i use `file` then `additionalFormData` wont be taken into account, but if i dont use file then i can use `additionalFormData`? With this i think we can call it `formData`, have the `file` string be part of it too.

Comment (fpeterdev) [inline src/platform/core/file/file-upload/README.md]: You are right, it's more straightforward that way.
To make that work `file` has to be optional to. What should happen when the user don't use neither `file` nor `code`? Throw an Error with the message "For [IUploadOptions] you have to set either the [file] or the [formData] property."?

Comment (emoralesb05) [inline src/platform/core/file/file-upload/README.md]: Yeah, we can though an error if both arent provided. 👍

**Comment by emoralesb05** [comment]
Awesome! Looks good, gonna merge this to include it in beta.4 😄 

Thanks dude!

**Pull request merged**